framework/src/meshgenerators/SideSetsFromBoundingBoxGenerator.C

```diff
@@ -80,6 +93,23 @@ SideSetsFromBoundingBoxGenerator::generate()
   bool found_side_sets = false;
   const bool inside = (_location == "INSIDE");
 
+  // Get the list of boundary ids from the boundary names
+  auto boundary_ids = MooseMeshUtils::getBoundaryIDs(*mesh, _boundaries_old, false);
+
+  // Check that the old boundary ids/names exist in the mesh
+  for (std::size_t i = 0; i < boundary_ids.size(); ++i)
+    if (boundary_ids[i] == Moose::INVALID_BOUNDARY_ID)
+      paramError(
+          "boundaries", "The boundary '", _boundaries_old[i], "' was not found within the mesh");
+
+  // Attempt to get the new boundary id from the name
+  auto boundary_id_new = MooseMeshUtils::getBoundaryID(_boundary_new, *mesh);
+
+  // If the new boundary id/name is not valid, make it instead
+  if (boundary_id_new == Moose::INVALID_BOUNDARY_ID)
+    boundary_id_new =
+        boundary_ids.empty() ? 0 : cast_int<boundary_id_type>(*boundary_ids.rbegin() + 1);
```

Comment:
why would boundary_ids be empty? Isnt it required?

also I dont get the logic of "last_value + 1". I think we need a whole new number if it s not valid. 
You should look at the MeshUtils for how it s done for subdomains (looks at all subdomain ids in the mesh, and takes the biggest + 1)

Reply:
Removing the empty check in next push. On the other thing, I don't understand what is confusing about it. If the boundary id number given for the name specified is invalid, it finds the last used boundary ID, and makes a new one with an ID right after, i.e. if the last boundary id is 2, it makes a boundary id of 3.

Reply:
it s not the last boundary ID in all the mesh. 
It s the last in your local context. You could be using a boundary ID that is used elsewhere

Reply:
will do
this conditional with a 0 is still an issue imo

Reply:
boundary_ids is only for the names you passed in, it s not for the entire mesh

Reply:
```std::vector<libMesh::boundary_id_type>
getBoundaryIDs(const libMesh::MeshBase & mesh,
               const std::vector<BoundaryName> & boundary_name,
               bool generate_unknown)
{
  const libMesh::BoundaryInfo & boundary_info = mesh.get_boundary_info();
  const std::map<libMesh::boundary_id_type, std::string> & sideset_map =
      boundary_info.get_sideset_name_map();
  const std::map<libMesh::boundary_id_type, std::string> & nodeset_map =
      boundary_info.get_nodeset_name_map();

  std::set<libMesh::boundary_id_type> boundary_ids = boundary_info.get_boundary_ids();
```
?

Reply:
maybe it does, I can't really tell, I will just change

Reply:
you passed a bunch of boundary names as an argument. You got the boundary ids for them. Not all boundary ids for all the mesh, that list got downsampled